docs/spec/qualifiers.rst

```diff
@@ -213,35 +213,44 @@ following should be allowed::
 Syntax
 ^^^^^^
 
-``Annotated`` is parameterized with a type and an arbitrary list of
-Python values that represent the annotations. Here are the specific
-details of the syntax:
+``Annotated`` is parameterized with a *base type* and an arbitrary list of
+Python values that represent *metadata* for the type::
 
-* The first argument to ``Annotated`` must be a valid type
+    from typing import Annotated
 
-* Multiple type annotations are supported (``Annotated`` supports variadic
+    Annotated[BaseType, Metadata1, Metadata2, ...]
```

Comment:
As I mentioned in discourse, I have misgivings about the term "base type" here. I'd prefer if we simply used the term "type". The problem with "base type" is that it implies that the type is modified by the metadata. First, this isn't the case for static type checkers. Second, metadata doesn't need to (and often doesn't in practice) refer to the type; it can apply to the symbol that the annotation is applied to — or something else completely.

Reply:
I think it's important for clarity to have some term that specifically refers to what I call the base type. Would using "inner type" address your concern?

Reply:
I don't think "inner type" is much better than "base type". I think it reads fine without giving it an adjective. `Annotated` allows one to specify a "type" and some metadata that may or may not have anything to do with the type.

Reply:
I would find it difficult to explain how PEP 746 works without some term specifically for the base type.

Reply:
Another option: We could call it the "base expression". That actually works better than type because the "base type" may be an annotation expression, not a type expression (e.g., `x: Annotated[Final[int], "some metadata"]` is legal).

Reply:
Yeah, I think "base expression" works well.

Reply:
Or "type expression", since this type argument needs to be a valid type expression, whereas the metadata expressions do not.

Reply:
> since this type argument needs to be a valid type expression

It doesn't; annotation expressions can be wrapped in Annotated[] according to the grammar I wrote. 

However, neither mypy nor pyright handle this the way I'd expect: https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKKYMAhjAKYAmANFAGKpkA2AUGwB4BchxpFGgG0mKVkNQwAuvQBEESuVlSoAXigBGNnF6jxkleq1c1UAEzbTFoA

```
from typing import Annotated, Final

x: Annotated[Final[int], "meta"] = 1
y: Final[int] = 1

x = 2  # no pyright error
y = 2  # pyright error
```

Pyright appears to simply ignore the Final, and mypy gives an error saying Final needs to be the outermost qualifier (https://mypy-play.net/?mypy=latest&python=3.12&gist=1069ad21b4b074710886b35744c9deaf). This is another discrepancy we'll have to resolve eventually.

Reply:
I like "inner type" or "wrapped type" for this. I agree with Eric in not particularly liking "base type"

Reply:
> annotation expressions can be wrapped in Annotated[] according to the grammar I wrote

Oh, I didn't catch that. I think that's a mistake. The first argument to `Annotated` must be a valid type. This is spelled out clearly later in the spec. The expression `Final[int]` is not a valid type, so it doesn't make sense for it to be used in `Annotated`. I think should be fixed this in the grammar, and type checkers should enforce this rule.

Reply:
I think the part of the spec that says that the first argument to `Annotated` must be a valid type is out-of-date. It looks like that restriction was relaxed in [this issue](https://bugs.python.org/issue46491). Elsewhere, the spec says that `Final` may be wrapped by `Annotated` (near the bottom of [this section](https://typing.readthedocs.io/en/latest/spec/qualifiers.html#semantics-and-examples)) and that `Required`, `NotRequired`, and `Annotated` can appear in any nesting order ([here](https://typing.readthedocs.io/en/latest/spec/typeddict.html#interaction-with-annotated)).

Reply:
Ah yes, I forgot about those provisions elsewhere in the spec. So yes, it sounds like we have an inconsistency here. I think it would be cleaner to allow only type expressions as the first type argument to `Annotated`, but I guess that ship has sailed, and it would be disruptive to undo that decision.

Reply:
I've updated pyright to honor `Final` when it's nested in `Annotated`: https://github.com/microsoft/pyright/issues/8097.